File: 04_Ejercicios/Ejercicios.java
public class Ejercicios{
public static void main(String[] args){
    fibonacci(8);
    System.out.println(factorial(5));
    System.out.println(palindroma("anitalavalatina"));
}

public static void fibonacci(int n){
    int a = 1;
    int b = 0; 
    if(n > 0){
    for(int i=0;i<n;i++){
        System.out.print(a+",");
        a = a + b;
        b = a - b;

    }
    }else{
        System.out.println("Por favor! Introducza un entero valido");
    }
    System.out.println("");
}


public static int factorial(int n){
    if(n == 0 || n == 1){
        return 1;
    }else{
        return n*factorial(n-1);
    }
}
public static boolean palindroma(String cadena){
    char[] char_array = new char[cadena.length()];
    int index = 0;
    for(int i=cadena.length()-1;i>=0;i--){
        char_array[index] = cadena.charAt(i);
        index++;     

    }

    String cadena2 = String.valueOf(char_array);
    
    return cadena.equals(cadena2);
    
    
}

}

File: 04_Ejercicios/Ejercicios.go
package main
import ("fmt";"strings")

func main(){
	fibonacci(8);
	fmt.Println(factorial(10));
	fmt.Println(palindroma("ot"));
}

func fibonacci(num int){
	a := 1;
	b := 0;
	for i:=0;i<num;i++ {
		fmt.Print(a);
		fmt.Print(",");
		a = a + b;
		b = a - b;
	}
	fmt.Println("");
}

func factorial(num int) int{
	if num == 0 || num == 1 { 
		return 1;
	}else {
		return num*factorial(num-1);
	}
}

func palindroma(cadena string) bool{
	str := strings.SplitAfter(cadena, "");
	var cadena2 string;
	for i:=len(str)-1;i>=0;i--{
		cadena2 += str[i];
	}

<<<<<<< HEAD
	return cadena == cadena2;
=======
	if(cadena2 == cadena){
		return true;
	}else{
		return false;
	}
>>>>>>> 01c3a036b65a10b525953f85cb6b981d3d27d4c8
    
	
	

}

